python-client/giskard/scanner/scanner.py

```diff
@@ -172,14 +171,17 @@ def _postprocess(self, issues: Sequence[Issue]) -> Sequence[Issue]:
 
     def _prepare_model_dataset(self, model: BaseModel, dataset: Optional[Dataset]):
         if model.is_text_generation and dataset is None:
-            logger.warning(
-                "No dataset provided. We will use TruthfulQA as a default dataset. This involves rewriting your model prompt."
-            )
-            from .llm.utils import load_default_dataset
-
-            dataset = load_default_dataset()
-            model = model.rewrite_prompt(
-                template="{question}", input_variables=["question"], feature_names=["question"]
+            logger.warning("No dataset provided. Inferring dataset")
+            from .llm.utils import infer_dataset
+
+            # TODO: remove this to use the datasets generated by BusinessDetector
```

Comment:
??? Not clear to me

Reply:
Actually I forgot to remove the `infer_dataset`. The idea is that each issue category generate a small dataset focused on the category inside the `BusinessDetector` and therefore it makes more sense to return the small datasets combined together instead of a more generic one that is not used in the scan anymore.